cdap-app-templates/cdap-etl/hydrator-spark-core-base/src/main/java/io/cdap/cdap/etl/spark/batch/SQLEngineRelationalEngine.java

```diff
@@ -67,8 +67,16 @@ public Engine getRelationalEngine() {
   public <T> Optional<SparkCollection<T>> tryRelationalTransform(
     StageSpec stageSpec, RelationalTransform transform, Map<String, SparkCollection<Object>> input) {
 
-    return adapter.tryRelationalTransform(stageSpec, transform, input).map(job ->
-      new SQLEngineCollection<T>(sec, functionCacheFactory, jsc, sqlContext,
-                                 datasetContext, sinkFactory, stageSpec.getName(), adapter, job));
+     return adapter.tryRelationalTransform(stageSpec, transform, input)
+      .map(job -> {
+        SQLEngineCollection<T> sqlEngineCollection = new SQLEngineCollection<T>(sec, functionCacheFactory, jsc,
+                                                                                sqlContext, datasetContext, sinkFactory,
+                                                                                stageSpec.getName(), adapter, job);
+        // If it is a Spark Sql Job with a local engine, then Perform a Pull resulting in RddCollection.
+        if (adapter.getSQLEngineClassName().equals(SparkSQLEngine.class.getName())) {
```

Comment:
How does this work if Pushdown is enabled? As far as I can tell, the adapter is not aware of the fallback engine.

Reply:
Ah, I see.